syntax = "proto3";

import "common.proto";
import "interface.proto";

package Net;

// The LAG service definition.
service VlanManagement {
    rpc CreateVlan (Vlan) returns (Result) {}
    rpc DeleteVlan (Vlan) returns (Result) {}

    rpc AddVlanMember (VlanMember) returns (Result) {}
    rpc RemoveVlanMember (VlanMember) returns (Result) {}
}

message VlanInstance {
    string id = 1;
}

enum VlanMode {
    VLAN_ACCESS = 0;
    VLAN_TAGGED = 1;
    VLAN_UNTAGGED = 2;
}

message Vlan {
    string id = 1;
}

message VlanMember {
    Vlan vlan = 1;
    VlanMode mode = 2;
    repeated Iface members = 3;
}
